src/compiler/scala/tools/nsc/transform/LambdaLift.scala

```diff
@@ -254,15 +254,26 @@ abstract class LambdaLift extends InfoTransform {
 
       afterOwnPhase {
         for ((owner, freeValues) <- free.toList) {
-          val newFlags = SYNTHETIC | (
-            if (owner.isClass) PARAMACCESSOR | PrivateLocal
-            else PARAM)
+          val newFlags = SYNTHETIC | (if (owner.isClass) PARAMACCESSOR else PARAM)
 
           proxies(owner) =
```

Comment:
I adopted part of your cleanup, but I like that the proxy is entered after the if, since you have to check `isClass` anyway in the `else` branch. Also changed the non-trait proxy's position to `fv.pos`. That sounds more logical, right?